swap/simulations_test.go

```diff
@@ -0,0 +1,752 @@
+			default:
+			}
+			// the node has all other peers in its peer list
+			if len(ts1.swap.peers) == 1 && len(ts2.swap.peers) == 1 {
+				break
+			}
+			// don't overheat the CPU...
+			time.Sleep(5 * time.Millisecond)
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		maxCheques := 42
+
+		p2Peer := ts1.peers[p2]
+		p1Peer := ts2.peers[p1]
+
+		for i := 0; i < maxCheques; i++ {
+			if i%2 == 0 {
+				p2Peer.Send(ctx, &testMsgBigPrice{})
+				err := waitForChequeProcessed(ts2)
+				if err != nil {
+					return err
+				}
+			} else {
+				p1Peer.Send(ctx, &testMsgBigPrice{})
+				err := waitForChequeProcessed(ts1)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		ch1, err := ts2.swap.loadLastReceivedCheque(p1)
+		if err != nil {
+			return err
+		}
+		ch2, err := ts1.swap.loadLastReceivedCheque(p2)
+		if err != nil {
+			return err
+		}
+
+		expected := uint64(maxCheques / 2 * (DefaultPaymentThreshold + 1))
+		if ch1.CumulativePayout != expected {
+			return fmt.Errorf("expected cumulative payout to be %d, but is %d", expected, ch1.CumulativePayout)
+		}
+		if ch2.CumulativePayout != expected {
+			return fmt.Errorf("expected cumulative payout to be %d, but is %d", expected, ch2.CumulativePayout)
+		}
+
+		return nil
+
+	})
+
+	if result.Error != nil {
+		t.Fatal(result.Error)
+	}
+
+	log.Info("Simulation ended")
+}
+
+// TestMultiChequeSimulation just launches two nodes, and sends multiple cheques
+// to the same node; checks that accounting still works properly afterwards and that
+// cheque cumulation values add up correctly
+func TestMultiChequeSimulation(t *testing.T) {
+	nodeCount := 2
+	// create the shared backend and params
+	params, err := newSharedBackendSwaps(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// cleanup backend
+	defer params.backend.Close()
+
+	// setup the wait for mined transaction function for testing
+	cleanup := setupContractTest()
+	defer cleanup()
+
+	params.backend.cashDone = make(chan struct{}, 1)
+	defer close(params.backend.cashDone)
+	// initialize the simulation
+	sim := simulation.NewInProc(newSimServiceMap(params))
+	defer sim.Close()
+
+	log.Info("Initializing")
+
+	ctx, cancelSimRun := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancelSimRun()
+
+	_, err = sim.AddNodesAndConnectFull(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	log.Info("starting simulation...")
+
+	result := sim.Run(ctx, func(ctx context.Context, sim *simulation.Simulation) (err error) {
+		log.Info("simulation running")
+
+		// define the nodes
+		debitor := sim.UpNodeIDs()[0]
+		creditor := sim.UpNodeIDs()[1]
+		// get the testService for the debitor
+		item, ok := sim.NodeItem(debitor, bucketKeySwap)
+		if !ok {
+			return errors.New("no swap in simulation bucket")
+		}
+		debitorSvc := item.(*testService)
+
+		// get the testService for the creditor
+		peerItem, ok := sim.NodeItem(creditor, bucketKeySwap)
```

Comment:
use `MustNodeItem`, it will save on the error checks and will panic if the item is not found

Reply:
Didn't know this one, reduces clutter